hoomd/md/compute.py

```diff
@@ -0,0 +1,344 @@
+        else:
+            return None
+
+    @log
+    def kinetic_energy(self):
+        """
+        :math:`K`, total kinetic energy of all particles in the group (in energy units).
+
+        .. math::
+
+            K = K_{\\mathrm{rot}} + K_{\\mathrm{trans}}
+
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_energy
+        else:
+            return None
+
+    @log
+    def translational_kinetic_energy(self):
+        """
+        :math:`K_{\\mathrm{trans}}`, translational kinetic energy of all particles in the group (in energy units).
+
+        .. math::
+
+            K_{\\mathrm{trans}} = \\frac{1}{2}\\sum_i m_i|\\vec{v}_i|^2
+
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.translational_kinetic_energy
+        else:
+            return None
+
+    @log
+    def rotational_kinetic_energy(self):
+        """
+        :math:`K_{\\mathrm{rot}}`, rotational kinetic energy of all particles in the group (in energy units).
+
+        Calculated as:
+
+        .. math::
+
+            K_{\\mathrm{rot}} = \\frac{1}{2} \\sum_i \\frac{L_i^2}{I_i},
+
+        where :math:`I` is the moment of inertia and :math:`L` is the angular momentum in the (diagonal) reference frame of the particle.
+
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.rotational_kinetic_energy
+        else:
+            return None
+
+    @log
+    def potential_energy(self):
+        """
+        :math:`U`, potential energy that the group contributes to the entire system (in energy units).
+
+        Calculated as:
+
+        .. math::
+
+            U = U_{\\mathrm{pair}} + U_{\\mathrm{bond}} + U_{\\mathrm{angle}} + U_{\\mathrm{dihedral}} + U_{\\mathrm{improper}},
+
+        where the pair term is calculated by summing the per-particle quantity :math:`U_{\\mathrm{pair}, i}` as
+
+        .. math::
+
+            U_{\\mathrm{pair}} = \\sum_{i \\in \\mathrm{filter}} U_{\\mathrm{pair, i}}
+
+            U_{\\mathrm{pair}, i} = \\frac{1}{2} \\sum_{j \\in \\mathrm{filter}} V_{\\mathrm{pair}, ij}.
+
+        The other terms are calculated as:
+
+        .. math::
+
+            U_{\\mathrm{bond}} = \\sum_{(i,j) \\in \\mathrm{bonds}} V_{\\mathrm{bond}, ij}
+
+            U_{\\mathrm{angle}} = \\sum_{(i,j,k) \\in \\mathrm{angles}} V_{\\mathrm{angle}, ijk},
+
+        with the rest calculated similarly.
+
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.potential_energy
+        else:
+            return None
+
+    @log
+    def degrees_of_freedom(self):
+        """
+        :math:`N_{\\mathrm{dof}}`, number of degrees of freedom given to the group by its integration method.
+
+        Note:
```

Comment:
For consistency with the rest of the docs here you can replace this note with an equation.